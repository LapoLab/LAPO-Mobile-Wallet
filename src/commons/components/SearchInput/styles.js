// @flow

export default {
  containerSearch: {
    marginVertical: 15,
    paddingHorizontal: 20,
  },
  inputSearch: {
    color: 'white',
    borderWidth: 1,
    fontSize: 18,
    paddingHorizontal: 10,
    paddingVertical: 5,
    height: 32,
    borderColor: 'white',
    borderRadius: 15,
    fontFamily: 'Lato-Regular',
  },
};
