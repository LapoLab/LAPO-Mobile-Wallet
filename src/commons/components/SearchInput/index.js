/**
 * @flow
 */

import React from 'react';
import {
  View,
  TextInput,
} from 'react-native';

import styles from './styles';

const SearchInput = ({ handleChange }: { handleChange: Function}) => (
  <View style={styles.containerSearch}>
    <TextInput style={styles.inputSearch} onChangeText={handleChange} />
  </View>
);

export default SearchInput;
