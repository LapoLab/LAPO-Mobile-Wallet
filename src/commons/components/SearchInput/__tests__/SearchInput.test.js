// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SearchInput from '../index';

describe('SearchInput component', () => {
  test('renders snapshots default', () => {
    const fn = jest.fn();
    const tree = renderer.create(<SearchInput handleChange={fn} />); // eslint-disable-line
    const render = tree.toJSON();
    expect(render).toMatchSnapshot();
  });
});
