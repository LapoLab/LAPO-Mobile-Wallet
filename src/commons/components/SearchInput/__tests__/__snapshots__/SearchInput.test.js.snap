// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component renders snapshots default 1`] = `
<View
  style={
    Object {
      "marginVertical": 15,
      "paddingHorizontal": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[MockFunction]}
    style={
      Object {
        "borderColor": "white",
        "borderRadius": 15,
        "borderWidth": 1,
        "color": "white",
        "fontFamily": "Lato-Regular",
        "fontSize": 18,
        "height": 32,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
